src/main/java/com/coveo/saml/XMLHelper.java

```diff
@@ -0,0 +1,109 @@
+package com.coveo.saml;
+
+import org.w3c.dom.DOMConfiguration;
+import org.w3c.dom.DOMImplementation;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.ls.DOMImplementationLS;
+import org.w3c.dom.ls.LSOutput;
+import org.w3c.dom.ls.LSSerializer;
+import org.w3c.dom.ls.LSSerializerFilter;
+
+import java.io.Writer;
+
+import java.util.Map;
+
+public class XMLHelper {
```

Comment:
This class is a utility class from OpenSAML 2 and I just copied the parts that SamlClient was using.

Reply:
We need to validate the licensing aspect of this... I think with Apache (which OpenSaml 2) uses we must at least retain the copyright headers on the file, when "borrowing" significant bits of the code.